src/main/java/backend/Logic.java

```diff
@@ -252,19 +250,31 @@ public Model getRepo(String repoId) {
      * @param labels The labels to be applied to the given issue
      * @return The list of labels on the issue after the request is received by GitHub
      */
-    public CompletableFuture<Boolean> replaceIssueLabels(TurboIssue issue, List<String> labels) {
-        logger.info(HTLog.format(issue.getRepoId(), "Applying labels " + labels + " to " + issue));
-
+    public CompletableFuture<Boolean> replaceIssueLabelsRepo(TurboIssue issue, List<String> labels) {
+        logger.info(HTLog.format(issue.getRepoId(), "Sending labels " + labels + " for " + issue + " to GitHub"));
         return repoIO.replaceIssueLabels(issue, labels).handle((resultLabels, ex) -> {
             if (ex == null) {
-                issue.setLabels(labels);
-                updateUIAndShow();
                 return true;
             } else {
+                undoUILabelChange(issue);
```

Comment:
Oh, right, yes. It should be reversed.
```java
return repoIO.replaceIssueLabels(issue, labels)
    .thenApply(newLabels -> true)
    .exceptionally(e -> {
        undoUILabelChange(issue);
        return false;
    });
```
I guess we don't need a `withResult`-like function for `thenApply`.